tests/test-path-manager.c

```diff
@@ -39,15 +40,23 @@ void test_pm_create(void const *test_data)
 
         assert(info->pm         != NULL);
         assert(info->pm->genl   != NULL);
-        assert(info->pm->id     != NULL);
-        assert(info->pm->family != NULL);
         assert(info->pm->nm     != NULL);
+
+        /*
+          Other struct mptcpd_pm fields may not have been initialized
+          yet since they depend on the existence of the "mptcpd"
+          generic netlink family.
+        */
 }
 
 void test_pm_destroy(void const *test_data)
 {
         struct test_info *const info = (struct test_info *) test_data;
 
+        if (!mptcpd_pm_ready(info->pm))
+                l_warn("Path manager was not ready.  "
```

Comment:
If the pm is not ready, is that an indication of a problem in the test, mptcpd, or the kernel? I think it would be good for this test to fail if a problem is detected.

There's the question of what to do if the kernel doesn't have the mptcpd genl family. Behave like ELL and skip the test? Or fail so it's clear the test environment was not set up for mptcpd?

Reply:
Is this case the path manager would not be ready if the `mptcp` genl family isn't available in the kernel.  Other than test the path manager life cycle, there isn't much else we can do with out the `mptcp` genl family.  This is particularly the case for the `test-commands` unit test since this pull request forces it to rely heavily on the existence of the `mptcp` genl family.  I don't think we can work around that with the new ELL genl API.

To be clear, I think the ELL genl API changes were good, and my comments above were not meant as a criticism.

Reply:
Hard fail sounds fine. If the distcheck case ends up too much of a headache in practice, we could adjust later.